runtime/services/catalog/artifacts/sql/sql.go

```diff
@@ -42,3 +51,69 @@ func (r *artifact) Serialise(ctx context.Context, catalogObject *drivers.Catalog
 	}
 	return catalogObject.GetModel().Sql, nil
 }
+
+var (
+	QueryCommentRegex     = regexp.MustCompile(`(?m)--.*$`)
+	MultipleSpacesRegex   = regexp.MustCompile(`\s\s+`)
+	SpacesAfterCommaRegex = regexp.MustCompile(`,\s+`)
+	MaterializedRegex     = regexp.MustCompile(`(?m)^[ \t]*--[ \t]?@materialize[ \t]?:[ \t]*([a-zA-Z]*)\s+`)
```

Comment:
What about not parsing whitespace before the comment? I can picture some users thinking they can materialize sub-queries, and at least we would catch that:

```
with x as (
  -- @materialize: true
  select ...
)
select ...;
```

So how about changing to: `(?m)^--[ \t]*@materialize[ \t]?:[ \t]*([a-zA-Z]*)\s+`? (also made whitespace after `--` a `*` instead of `?`)

Reply:
hmm it can still be confusing, consider  -
1. In the example you have given, after making the suggested changes, we will not match and thus materialize will be unspecified so there won't be any error and tag will not have any effect but user may think it materialized as there was no error.
2. In the example below, we will match but materialize entire query not just inner query which may be the intent of the user as you said.
```
with x as (
-- @materialize: true
  select ...
)
select ...;
```
So what about restricting the tag only at the first line of the model and not doing multi line (as per your original spec in notion) using `^[ \t]*--[ \t]*@materialize[ \t]?:[ \t]*([a-zA-Z]*)\s+`. We can have one more regex which checks for this tag not in the first line and throw error saying `materialize tag only allowed at first line`. What do you think ?

Reply:
Agree with your observations. I think there might be some cases where requiring first line isn't great. Someone might want to put a license or other comment there, or we might add more `@param: value` annotations soon. 

Doing an extra check for errors sounds nice in theory, but might be redundant complexity (also, this runs on every keystroke when modeling). This doesn't need to be perfect, so I wouldn't worry about it.

IMO, doing multiline search with `^--[ \t]*@materialize[ \t]*:[ \t]*([a-zA-Z]*)\s+` will solve most cases without matching too many unintended cases.

Reply:
ok

Reply:
pushed the change